core/src/toga/icons.py

```diff
@@ -1,34 +1,26 @@
 from __future__ import annotations
 
-import sys
 import warnings
 from pathlib import Path
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Callable, Iterable, Union
 
 import toga
 from toga.platform import get_platform_factory
+from toga.types import TypeAlias
 
 if TYPE_CHECKING:
-    if sys.version_info < (3, 10):
-        from typing_extensions import TypeAlias
-    else:
-        from typing import TypeAlias
-
-    IconContent: TypeAlias = str | Path | toga.Icon
+    IconContent: TypeAlias = Union[str, Path, toga.Icon]
```

Comment:
This seems odd -I seem to recall we were getting something out of `typing_extensions` that we didn't get out of the base `typing` versions of TypeAlias... but I can't work out what that is any more. It's not a link to the type, because that's not currently resolving as a link in the docs...

Reply:
Writing `Union`s like this wasn't supported until Python 3.10....so, mypy complains about it.

Reply:
I thought there was something more about TypeAlias itself... I could be misremembering though.